aeppl/transforms.py

```diff
@@ -66,76 +34,218 @@ def log_jac_det(self, value: TensorVariable, *inputs) -> TensorVariable:
+        base_op = clsdict.get("base_op", None)
+        default = clsdict.get("default", False)
+
+        if default and base_op is not None:
+            # Create dispatch functions
+            @_default_transformed_rv_op.register(type(base_op))
+            def default_transformed_rv_op(op):
+                return cls_res()
+
+        return cls_res
+
+
+class TransformedRV(RandomVariable, metaclass=TransformedRVMeta):
     r"""A base class for transformed `RandomVariable`\s."""
 
 
-@_logprob.register(TransformedRV)
-def transformed_logprob(op, value, *inputs, name=None, **kwargs):
+@singledispatch
+def _default_transformed_rv_op(
+    op: Op,
+) -> Optional[Type[TransformedRV]]:
+    """Return a default ``TransformedRV`` for a given ``RandomVariable`` ``Op``.
+
+    This function dispatches on the type of ``Op``, which should be a subclass
+    of ``RandomVariable``.  If you want to implement new transforms for a
+    ``RandomVariable``, register a function on this dispatcher.
+
     """
-    Compute logp graph for a value variable that was back-transformed to be on
-    the natural support of the respective random variable.
+    return None
+
+
+def create_transformed_rv_op(
+    rv_op: Op,
+    transform: Transform,
+    *,
+    default: bool = False,
+    cls_dict_extra: Optional[Dict] = None,
+) -> Type[TransformedRV]:
+    """Create a new ``TransformedRV`` given a base ``RandomVariable`` ``Op``
+
+    Pass `default = True` to specify that the returned ``TransformedRV`` should
+    be the default transformation for the base ``RandomVariable``
+
     """
 
-    logprob = _logprob(op.base_op, value, *inputs, name=name, **kwargs)
+    trans_name = getattr(transform, "name", "transformed")
+    rv_type_name = type(rv_op).__name__
+    cls_dict = type(rv_op).__dict__.copy()
+    rv_name = cls_dict.get("name", "")
+    if rv_name:
+        cls_dict["name"] = f"{rv_name}_{trans_name}"
+    cls_dict["base_op"] = rv_op
+    cls_dict["transform"] = transform
+    cls_dict["default"] = default
 
-    original_forward_value = op.transform.forward(value, *inputs)
-    jacobian = op.transform.log_jac_det(original_forward_value, *inputs)
+    if cls_dict_extra is not None:
+        cls_dict.update(cls_dict_extra)
 
-    if name:
-        logprob.name = f"{name}_logprob"
-        jacobian.name = f"{name}_logprob_jac"
+    new_op_type = type(f"Transformed{rv_type_name}", (TransformedRV,), cls_dict)
 
-    return logprob + jacobian
+    return new_op_type
+
+
+DEFAULT_TRANSFORM = object()
+
+
+class TransformValuesMapping(Feature):
+    r"""A `Feature` that maintains a map between value variables and their transforms."""
+
+    def __init__(self, values_to_transforms):
+        self.values_to_transforms = values_to_transforms
+
+    def on_attach(self, fgraph):
+        if hasattr(fgraph, "transform_values_mapping"):
+            raise ValueError(
+                f"{fgraph} already has the `TransformValuesMapping` feature attached."
```

Comment:
Is it worth it to add a test for this `ValueError`? There is something similar in the `PreserveRVMappings` that is also not being tested AFAIK. 

Also, on the Aesara docs, it suggests we should raise the more specific `AlreadyThere` exception: https://aesara.readthedocs.io/en/latest/library/graph/fgraph.html?highlight=feature#aesara.graph.features.Feature.on_attach

Should we do that?

Reply:
Yeah, the `AlreadyThere` approach sounds better (assuming it really was/is fully supported).